roxie/roxiemem/roxiemem.cpp

```diff
@@ -218,6 +221,73 @@ inline void notifyMemoryUnused(void * address, memsize_t size)
 #endif
 }
 
+extern void notifyAllUnusedRoxieMem();
+
+extern void lockRoxieMem(bool lock)
+{
+#ifndef _WIN32
+    if (!heapBase || !heapEnd)
+    {
+        DBGLOG("Attempt to lock heap memory failed, invalid heapBase or heapEnd");
+        return;
+    }
+
+    memsize_t memsize = heapEnd - heapBase;
+
+    if (lock && !heapLocked)
+    {
+        // mlock() should fault in all pages ...
+        int srtn = mlock(heapBase, memsize);
+        if (srtn)
```

Comment:
comment:  Not something that needs to be changed.

In general I would recommend putting the likely, non-error condition first in a conditional statement (if there is an else clause).  It is easier to read, because as you read top to bottom you follow the natural flow of the program.  With the error case first I need to jump to the else to follow the normal behaviour.